zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,697 @@
+    deltas : Expr or 'auto', optional
+        The expression to use for the point in time adjustments.
+        If the string 'auto' is passed, a deltas expr will be looked up
+        by stepping up the expression tree and looking for another field
+        with the name of ``expr`` + '_deltas'. If None is passed, no deltas
+        will be used.
+    loader : BlazeLoader, optional
+        The blaze loader to attach this pipeline dataset to. If none is passed,
+        the global blaze loader is used.
+    resources : dict or any, optional
+        The data to execute the blaze expressions against. This is used as the
+        scope for ``bz.compute``.
+    no_deltas_rule : {'warn', 'raise', 'ignore'}
+        What should happen if ``deltas='auto'`` but no deltas can be found.
+        'log' says to log a message but continue.
+        'raise' says to raise an exception if no deltas can be found.
+        'ignore' says take no action and proceed with no deltas.
+
+    Returns
+    -------
+    pipeline_api_obj : DataSet or BoundColumn
+        Either a new dataset or bound column based on the shape of the expr
+        passed in. If a tabular shaped expression is passed, this will return
+        a ``DataSet`` that represents the whole table. If an array-like shape
+        is passed, a ``BoundColumn`` on the dataset that would be constructed
+        from passing the parent is returned.
+    """
+    deltas = _get_deltas(expr, deltas, no_deltas_rule)
+    if deltas is not None:
+        invalid_nodes = tuple(filter(
+            complement(flip(isinstance, valid_deltas_node_types)),
+            expr._subterms(),
+        ))
+        if invalid_nodes:
+            raise TypeError(
+                'expression with deltas may only contain (%s) nodes,'
+                " found: %s" % (
+                    ', '.join(map(getname, valid_deltas_node_types)),
+                    ', '.join(map(compose(getname, type), invalid_nodes)),
+                ),
+            )
+
+    # Check if this is a single column out of a dataset.
+    single_column = None
+    if isscalar(expr.dshape.measure):
+        # This is a single column, record which column we are to return
+        # but create the entire dataset.
+        single_column = expr._name
+        col = expr
+        for expr in expr._subterms():
+            if isrecord(expr.dshape.measure):
+                break
+        else:
+            expr = bz.Data(col, name=single_column)
+
+    measure = expr.dshape.measure
+    if not isrecord(measure) or AD_FIELD_NAME not in measure.names:
+        raise TypeError(
+            "expr must be a collection of records with at least an '{ad}'"
+            " field. Fields provided: '{fields}'\nhint: maybe you need to use "
+            ' `relabel` to change your field names'.format(
+                ad=AD_FIELD_NAME,
+                fields=measure,
+            ),
+        )
+    _check_datetime_field(AD_FIELD_NAME, measure)
```

Comment:
Every time I end up relying on global constants like this, I end up wanting to change it later and regretting how hard it is to do so.  How hard would it be to make these just default parameters to this function.  Are they used in multiple other places?